### ISSUE TITLE
Provider crashes when used in container

### ISSUE THREAD

**Issue by tcassaert** [rancher/terraform-provider-rke#235]
I'm trying to do a `terraform apply` from within the official Hashicorp Terraform container image.

My repository looks like the following:
```
.
├── rke_openstack
│   ├── modules
│   │   ├── compute
│   │   │   ├── main.tf
│   │   │   ├── outputs.tf
│   │   │   └── variables.tf
│   │   ├── ips
│   │   │   ├── main.tf
│   │   │   ├── outputs.tf
│   │   │   └── variables.tf
│   │   ├── network
│   │   │   ├── main.tf
│   │   │   ├── outputs.tf
│   │   │   └── variables.tf
│   │   └── rke
│   │       ├── main.tf
│   │       ├── outputs.tf
│   │       └── variables.tf
│   ├── README.md
│   ├── rke.tf
│   ├── rke.tfvars
│   ├── terraform.d
│   │   └── plugins
│   │       └── linux_amd64
│   │           └── terraform-provider-rke
│   ├── variables.tf
│   └── versions.tf
```

I run Terraform like this:

```
podman run -it --env-file .openstack_env --volume $(pwd):/terraform -w /terraform/rke_openstack --entrypoint='' hashicorp/terraform:0.12.28 sh -c "TF_VAR_os_project_id=\$OS_PROJECT_ID TF_VAR_os_username=\$OS_USERNAME TF_VAR_os_domain_name=\$OS_USER_DOMAIN_NAME TF_VAR_os_password=\$OS_PASSWORD TF_VAR_os_auth_url=\$OS_AUTH_URL terraform apply -var-file=/terraform/rke_openstack/rke.tfvars"
```

The env variables are OpenStack credentials that are necessary for the cloud_provider part.

This gives me the following:

```
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: [signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x1a720e3]                                                                                                    
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:                                                       
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: goroutine 101 [running]:                              
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: github.com/rancher/terraform-provider-rke/rke.readClusterState(0xc0003c5960, 0x0, 0x0, 0x0)                                                                                 
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/src/github.com/rancher/terraform-provider-rke/rke/resource_rke_cluster.go:265 +0x283                                                                            
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: github.com/rancher/terraform-provider-rke/rke.resourceRKEClusterRead(0xc0003c5960, 0x1ce48e0, 0xc0005fa9f0, 0xc0003c5960, 0x0)                                              
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/src/github.com/rancher/terraform-provider-rke/rke/resource_rke_cluster.go:98 +0xdf                                                                              
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000131b90, 0xc000aacbe0, 0x1ce48e0, 0xc0005fa9f0, 0xc00000fdd8, 0x0, 0x0)      
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/helper/schema/resource.go:460 +0x119                                                                  
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc0000c66f0, 0x23d1760, 0xc0009a0030, 0xc0001683c0, 0xc0000c66f0, 0xc0009a0030, 0xc0009f0b30)                                                                                                
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/internal/helper/plugin/grpc_provider.go:525 +0x3d8                                                    
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x1ef6400, 0xc0000c66f0, 0x23d1760, 0xc0009a0030, 0xc000168360, 0x0, 0x23d1760, 0xc0009a0030, 0xc000d7c000, 0x2cc0f)                                                                                  
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/internal/tfplugin5/tfplugin5.pb.go:3269 +0x217                                                        
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004d9380, 0x23f6080, 0xc000562600, 0xc000175500, 0xc000cd2510, 0x340a310, 0x0, 0x0, 0x0)                               
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:1024 +0x4f4                                                                                                    
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: google.golang.org/grpc.(*Server).handleStream(0xc0004d9380, 0x23f6080, 0xc000562600, 0xc000175500, 0x0)                                                                     
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:1313 +0xd97                                                                                                    
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0004b18a0, 0xc0004d9380, 0x23f6080, 0xc000562600, 0xc000175500)
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:722 +0xbb
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-07-30T16:31:32.168Z [DEBUG] plugin.terraform-provider-rke:         /go/pkg/mod/google.golang.org/grpc@v1.27.1/server.go:720 +0xa1
2020-07-30T16:31:32.171Z [DEBUG] plugin: plugin process exited: path=/terraform/rke_openstack/terraform.d/plugins/linux_amd64/terraform-provider-rke pid=238 error="exit status 2"
2020-07-30T16:31:32.171Z [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2020/07/30 16:31:32 [ERROR] module.rke: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2020/07/30 16:31:32 [ERROR] module.rke: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
```
When running the same command, not in a container, the `terraform apply` works as expected.

```
TF_VAR_os_project_id=$OS_PROJECT_ID TF_VAR_os_username=$OS_USERNAME TF_VAR_os_domain_name=$OS_USER_DOMAIN_NAME TF_VAR_os_password=$OS_PASSWORD TF_VAR_os_auth_url=$OS_AUTH_URL terraform apply -var-file rke.tfvars
```

### Terraform version
0.12.28 in both cases

### Provider version
v1.0.1 in both cases

**Comment by rawmind0** [comment]
SIGSEV should be addressed on PR #237

**Comment by rawmind0** [comment]
PR merged into master. Please, reopen issue if needed.
